src/jit/lowerarmarch.cpp

```diff
@@ -280,93 +253,68 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
             // it to a larger constant whose size is sufficient to support
             // the largest width store of the desired inline expansion.
 
-            ssize_t fill = initVal->gtIntCon.gtIconVal & 0xFF;
+            ssize_t fill = src->AsIntCon()->IconValue() & 0xFF;
+
             if (fill == 0)
             {
-                MakeSrcContained(blkNode, source);
+                src->SetContained();
             }
-            else if (size < REGSIZE_BYTES)
+            else if (size >= REGSIZE_BYTES)
             {
-                initVal->gtIntCon.gtIconVal = 0x01010101 * fill;
+                fill *= 0x0101010101010101LL;
+                src->gtType = TYP_LONG;
             }
             else
             {
-                initVal->gtIntCon.gtIconVal = 0x0101010101010101LL * fill;
-                initVal->gtType             = TYP_LONG;
+                fill *= 0x01010101;
             }
-            blkNode->gtBlkOpKind = GenTreeBlk::BlkOpKindUnroll;
+
+            src->AsIntCon()->SetIconValue(fill);
         }
         else
 #endif // _TARGET_ARM64_
         {
+            // TODO-ARM-CQ: Currently we generate a helper call for every initblk we encounter.
+            // Later on we should implement loop unrolling code sequences to improve CQ.
             blkNode->gtBlkOpKind = GenTreeBlk::BlkOpKindHelper;
         }
     }
     else
     {
-        // CopyObj or CopyBlk
-        // Sources are src and dest and size if not constant.
+        assert(src->OperIs(GT_IND, GT_LCL_VAR, GT_LCL_FLD));
+        src->SetContained();
 
-        if (blkNode->OperGet() == GT_STORE_OBJ)
+        if (src->OperIs(GT_IND))
         {
-            // CopyObj
-            GenTreeObj* objNode = blkNode->AsObj();
-
-            unsigned slots = objNode->GetLayout()->GetSlotCount();
-
-#ifdef DEBUG
-            // CpObj must always have at least one GC-Pointer as a member.
-            assert(objNode->GetLayout()->HasGCPtr());
```

Comment:
Ah, nice catch. This change shouldn't probably be in this PR, though it's small and doesn't hurt.

One of the removed commits (https://github.com/mikedn/coreclr/commit/8336b448d407d3e807007f12cb978017f6cb5d88) is moving the code from `fgMorphUnsafeBlk` to lowering. That's responsible for changing OBJs without GC pointers into BLKs.

I'm also working on #27053 that will make BLKs even more rare. Since introducing `ClassLayout` and making `OBJ` a small node there's no longer any need to discard struct type information from IR. We should use OBJ whenever type information is available, with or without GC pointers.

Reply:
Yes, currently `GT_OBJ` nodes always have GC pointers. The plan is to use `GT_OBJ` nodes even in cases when no GC pointers are present, at least until lowering. After lowering it probably doesn't matter anymore.

Reply:
Makes sense - in that case perhaps we don't need to change it.